### PR TITLE
fix the delete comment with replies

### PR DESCRIPTION
**Pull request by ConstantineMaliza**
**What does this PR do?**

- the user should able to delete request and comment left on that request must delete directly and reply

**Description of Task to be completed?**

- The endpoint  **http://localhost:5000/request/requestId/comment/commentId**
- http://localhost:5000/requests/requestId

### CODE REVIEW

**Comment by Anguandia** [comment]
@SimonImanigirimpuhwe, @SamuelSabasNayo, and @joshwambere please review this fix ASAP

**Review by SimonImanigirimpuhwe — CHANGES_REQUESTED** [review]
I found that this doesn't fix issue 100% as expected... You can delete a comment and its reply get deleted but when you delete a request all comments get deleted but replies remain. Plz work on that

File: src/controllers/comment.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import model from "../database/models";
+import model from '../database/models';
 
 export const CreateComment = async (req, res) => {
```
Comment (SamuelSabasNayo) [inline src/controllers/comment.js:3]: Anyone can add a comment on any request?
Or it is between **Manger** and **Requester**.

File: src/controllers/comment.js [hunk 2]
```diff
@@ -96,53 +92,53 @@ export const ReplyComment = async (req, res) => {
...
     res.status(200).json({
       reply,
     });
   } catch (error) {
-    res.status(404).json({message:error.message})
+    res.status(404).json({ message: error.message });
   }
 };
 
-export const DeleteComment = async(req, res) => {
```
Comment (SamuelSabasNayo) [inline src/controllers/comment.js:137]: Again, anyone can delete someone's else comment.

**Review by SamuelSabasNayo — CHANGES_REQUESTED** [review]

File: src/controllers/comment.js (hunk 1 above)
Comment (ConstantineMaliza) [reply src/controllers/comment.js:3]: no, you need to specify requestId

**Comment by Anguandia** [comment]
@joshwambere please review this PR

**Review by joshwambere — COMMENTED** [review]
I'm having a socket hanging issue with userData.id` when trying to delete comment

**Review by thepatrickniyo — COMMENTED** [review]
A rebase is needed since onother work was merged

**Review by SimonImanigirimpuhwe — APPROVED** [review]
Good implementation

**Review by SamuelSabasNayo — APPROVED** [review]
Good work @ConstantineMaliza!

**Review by kabano89 — APPROVED** [review]

**Review by thepatrickniyo — APPROVED** [review]

**Review by joshwambere — APPROVED** [review]

**Pull request merged**